BrowserKit/Sources/MenuKit/MenuTableView.swift

```diff
@@ -81,11 +85,15 @@ class MenuTableView: UIView, UITableViewDataSource, UITableViewDelegate {
         _ tableView: UITableView,
         didSelectRowAt indexPath: IndexPath
     ) {
-        guard let action = menuData[indexPath.section].options[indexPath.row].action else {
-            tableView.deselectRow(at: indexPath, animated: true)
-            return
+        tableView.deselectRow(at: indexPath, animated: false)
+
+        if let action = menuData[indexPath.section].options[indexPath.row].action {
+            action()
         }
-        action()
-        tableView.deselectRow(at: indexPath, animated: true)
+    }
+
+    func reloadTableView(with data: [MenuSection]) {
+        menuData = data
+        tableView.reloadData()
```

Comment:
do we want to consider using diffable data source, not sure on the implications here, but for the homepage and the tab tray, we were reloading the whole table view instead of using snapshots. This is a problem because we are reloading so many times. I don't see any issues since I believe the reload doesn't happen as often, but thought I'll make a note if that's something we're considering or would want to.

Reply:
I thought about it, but, honestly, I think that's just ultra optimization in this instance. The view is so small, the options so few, and the reloads so rare that I don't feel like we'd gain anything truly meaningful from doing it that way.